core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -154,7 +158,7 @@ private[storage] class BlockManagerDecommissioner(
   // Shuffles which are queued for migration & number of retries so far.
   // Visible in storage for testing.
   private[storage] val shufflesToMigrate =
-    new java.util.concurrent.ConcurrentLinkedQueue[(ShuffleBlockInfo, Int)]()
+    new java.util.concurrent.LinkedBlockingQueue[(ShuffleBlockInfo, Int)]()
```

Comment:
This seems like it's adding locking overhead compared to the concurrentlinkedqueue, can you elaborate on this choice?

With ConcurrentLinkedQueue the producers have contention and the consumers have no contention (from https://www.baeldung.com/java-queue-linkedblocking-concurrentlinked ) whereas with the LinkedBlockingQueue the consumers are going to have lock contention more generally I believe.

Reply:
Thanks for pointing this out.

AFAIK, both `ConcurrentLinkedQueue` and `LinkedBlockingQueue` are guaranteed to ensure thread-safe for their operations. And `ConcurrentLinkedQueue` is based on CAS while `LinkedBlockingQueue` is based on `ReentrantLock`. So, `ConcurrentLinkedQueue` is generally faster for `LinkedBlockingQueue`.

And in this specific case, using `ConcurrentLinkedQueue` requires us to manually maintain a periodic(per second) checking for the queue, which may occupy unnecessary CPU resources for a while when the queue is empty (e.g., after the first round migration)...but, it should not be big deal since the node is going to shut down...

And for `LinkedBlockingQueue`, because of the blocking functionality, we 1) don't need the manual checking, which simplifies the logic 2) won't occupy CPU resources when the queue is empty

And if we care more about the reading performance of the queue, I'm ok to revert this to use the `ConcurrentLinkedQueue`. WDYT?

Reply:
I think we generally care more about read performance of this data structure. In the situation where we are busy waiting we don't really care because there isn't any data for us to migrate and the executor is exiting.

Reply:
Ok, then I'll revert this part of change.